Mathlib/Order/CompleteLatticeIntervals.lean

```diff
@@ -0,0 +1,169 @@
+default values for `supₛ` and `infₛ`.
+-/
+
+
+open Classical
+
+open Set
+
+variable {α : Type _} (s : Set α)
+
+section SupSet
+
+variable [SupSet α]
+
+/-- `SupSet` structure on a nonempty subset `s` of an object with `SupSet`. This definition is
+non-canonical (it uses `default s`); it should be used only as here, as an auxiliary instance in the
+construction of the `ConditionallyCompleteLinearOrder` structure. -/
+noncomputable def subsetSupSet [Inhabited s] : SupSet s where
+  supₛ t :=
+    if ht : supₛ ((↑) '' t : Set α) ∈ s
+    then ⟨supₛ ((↑) '' t : Set α), ht⟩
+    else default
+#align subset_has_Sup subsetSupSet
+
+attribute [local instance] subsetSupSet
+
+@[simp]
+theorem subset_supₛ_def [Inhabited s] :
+    @supₛ s _ = fun t =>
+      if ht : supₛ ((↑) '' t : Set α) ∈ s
+      then ⟨supₛ ((↑) '' t : Set α), ht⟩
+      else default :=
+  rfl
+#align subset_Sup_def subset_supₛ_def
+
+theorem subset_supₛ_of_within [Inhabited s] {t : Set s} (h : supₛ ((↑) '' t : Set α) ∈ s) :
+    supₛ ((↑) '' t : Set α) = (@supₛ s _ t : α) := by simp [dif_pos h]
+#align subset_Sup_of_within subset_supₛ_of_within
+
+end SupSet
+
+section InfSet
+
+variable [InfSet α]
+
+/-- `InfSet` structure on a nonempty subset `s` of an object with `InfSet`. This definition is
+non-canonical (it uses `default s`); it should be used only as here, as an auxiliary instance in the
+construction of the `ConditionallyCompleteLinearOrder` structure. -/
+noncomputable def subsetInfSet [Inhabited s] : InfSet s where
+  infₛ t :=
+    if ht : infₛ ((↑) '' t : Set α) ∈ s
+    then ⟨infₛ ((↑) '' t : Set α), ht⟩
+    else default
+#align subset_has_Inf subsetInfSet
+
+attribute [local instance] subsetInfSet
+
+@[simp]
+theorem subset_infₛ_def [Inhabited s] :
+    @infₛ s _ = fun t =>
+      if ht : infₛ ((↑) '' t : Set α) ∈ s
+      then ⟨infₛ ((↑) '' t : Set α), ht⟩ else
+      default :=
+  rfl
+#align subset_Inf_def subset_infₛ_def
+
+theorem subset_infₛ_of_within [Inhabited s] {t : Set s} (h : infₛ ((↑) '' t : Set α) ∈ s) :
+    infₛ ((↑) '' t : Set α) = (@infₛ s _ t : α) := by simp [dif_pos h]
+#align subset_Inf_of_within subset_infₛ_of_within
+
+end InfSet
+
+variable [ConditionallyCompleteLinearOrder α]
+
+attribute [local instance] subsetSupSet
+
+attribute [local instance] subsetInfSet
+
+set_option linter.unusedVariables false in --Porting note: Variables are actually used - linter bug?
+/-- For a nonempty subset of a conditionally complete linear order to be a conditionally complete
+linear order, it suffices that it contain the `supₛ` of all its nonempty bounded-above subsets, and
+the `infₛ` of all its nonempty bounded-below subsets.
+See note [reducible non-instances]. -/
+@[reducible]
+noncomputable def subsetConditionallyCompleteLinearOrder [Inhabited s]
+    (h_Sup : ∀ {t : Set s} (_ : t.Nonempty) (h_bdd : BddAbove t), supₛ ((↑) '' t : Set α) ∈ s)
+    (h_Inf : ∀ {t : Set s} (_ : t.Nonempty) (h_bdd : BddBelow t), infₛ ((↑) '' t : Set α) ∈ s) :
```

Comment:
```suggestion
/-- For a nonempty subset of a conditionally complete linear order to be a conditionally complete
linear order, it suffices that it contain the `supₛ` of all its nonempty bounded-above subsets, and
the `infₛ` of all its nonempty bounded-below subsets.
See note [reducible non-instances]. -/
@[reducible]
noncomputable def subsetConditionallyCompleteLinearOrder [Inhabited s]
    (h_Sup : ∀ {t : Set s} (_ : t.Nonempty) (_h_bdd : BddAbove t), supₛ ((↑) '' t : Set α) ∈ s)
    (h_Inf : ∀ {t : Set s} (_ : t.Nonempty) (_h_bdd : BddBelow t), infₛ ((↑) '' t : Set α) ∈ s) :
```
I am not sure this is the correct solution, but it makes the linter happy, and it is what Scott does in his video.

Reply:
In any case LGTM.

bors d+